tabs/utils/service-control.sh

```diff
@@ -0,0 +1,273 @@
+#!/bin/sh -e
+
+# Load common script functions
+. ../common-script.sh  
+
+#external services directory 
+SCRIPT_DIR="./services"
+
+# Function to show the main menu
+show_menu() {
+    clear
+    echo "============================"
+    echo " Service Management Menu"
+    echo "============================"
+    echo "1. View all services"
+    echo "2. View enabled services"
+    echo "3. View disabled services"
+    echo "4. Add a new service"
+    echo "5. Remove a service"
+    echo "6. Start a service"
+    echo "7. Stop a service"
+    echo "8. Enable a service"
+    echo "9. Disable a service"
+    echo "10. Create a service from external scripts"
+    echo "11. Exit"
+    echo "============================"
+}
+
+# Function to view all services
+view_all_services() {
+    echo "Listing all services..."
+    $ESCALATION_TOOL systemctl list-units --type=service --all --no-legend | awk '{print $1}' | sed 's/\.service//' | more
+}
+
+# Function to view enabled services
+view_enabled_services() {
+    echo "Listing enabled services..."
+    $ESCALATION_TOOL systemctl list-unit-files --type=service --state=enabled --no-legend | awk '{print $1}' | sed 's/\.service//' | more
+}
+
+# Function to view disabled services
+view_disabled_services() {
+    echo "Listing disabled services..."
+    $ESCALATION_TOOL systemctl list-unit-files --type=service --state=disabled --no-legend | awk '{print $1}' | sed 's/\.service//' | more
+}
+
+# Function to view started services
+view_started_services() {
+    echo "Listing started services:"
+    $ESCALATION_TOOL systemctl list-units --type=service --state=running --no-pager | head -n -6 | awk 'NR>1 {print $1}' | more
+}
+
+# Function to add a new service
+add_service() {
+    while [ -z "$SERVICE_NAME" ]; do
+        echo "Enter the name of the new service (e.g., my_service):"
+        read -r SERVICE_NAME
+    done    
```

Comment:
Should check if service already exists. Maybe something like this?

```suggestion
    while [ -z "$SERVICE_NAME" ]; do
        echo "Enter the name of the new service (e.g., my_service):"
        read -r SERVICE_NAME

        if $ESCALATION_TOOL systemctl list-units --type=service --all --no-legend | grep -q "$SERVICE_NAME.service"; then
            echo "Service already exists!"
            SERVICE_NAME=""
        fi
    done
```

Reply:
is it working code?

Reply:
Technically, yes.